### ISSUE TITLE
App not loading

### ISSUE THREAD

**Issue by yocarry** [TravisLedo/LidarScanRender#2]
When I try to run the java renderer application from the command line it spits out this error every time: `Error: Could not find or load main class lidarscanrender.Main
Caused by: java.lang.NoClassDefFoundError: javafx/application/Application`

**Comment by TravisLedo** [comment]
> When I try to run the java renderer application from the command line it spits out this error every time: `Error: Could not find or load main class lidarscanrender.Main Caused by: java.lang.NoClassDefFoundError: javafx/application/Application`

Have you tried running it from an IDE? What was the command you used?

**Comment by yocarry** [comment]
The command was: java -jar "C:\Users\Admin\Desktop\TravisLedo-LidarScanRender-44895d9\TravisLedo-LidarScanRender-44895d9\LidarScanRender\dist\LidarScanRender.jar"

I have tried running it from the Apache NetBeans IDE, but it failed to load the build. I have tried running this app for hours. I am using JDK 17.0.2 by the way.

Also, thanks for responding fast, I really appreciate it!

Full Errors from the IDE:
https://github.com/TravisLedo/LidarScanRender/files/7943739/errors.txt

**Comment by TravisLedo** [comment]
> The command was: java -jar "C:\Users\Admin\Desktop\TravisLedo-LidarScanRender-44895d9\TravisLedo-LidarScanRender-44895d9\LidarScanRender\dist\LidarScanRender.jar"
> 
> I have tried running it from the Apache NetBeans IDE, but it failed to load the build. I have tried running this app for hours. I am using JDK 17.0.2 by the way.
> 
> Also, thanks for responding fast, I really appreciate it!
> 
> Full Errors from the IDE: https://github.com/TravisLedo/LidarScanRender/files/7943739/errors.txt


Hey I just tested it myself on my new computer. I think I know what the errors are because I ran into the same issue but got it working. You have to import the jar files into the libraries.

In Netbeans on my Mac I did that by going to Tools>Libraries>NewLibrary>Name it something like "JavaFX_17">Add Jar/Folder.

Then back in the Projects hierarchy view where you can select which file to work on, right click on Libraries>Add Libraries>Select the "JavaFX_17" library you just added.

You will need to do this for JSerialComm as well, that jar file is included in this repo. The JavaFX jars you will have to download from here https://gluonhq.com/products/javafx/
The files you need to import are in the lib folder of the download.

Instructions with pictures can be found here https://openjfx.io/openjfx-docs/ 
Click on JavaFX and Netbeans on the left.

Let me know if that gets you any closer.

**Comment by yocarry** [comment]
Update: All the errors are still there

**Comment by TravisLedo** [comment]
> `Error occurred during initialization of boot layer java.lang.module.FindException: Module javafx.controls not found C:\Users\Admin\Desktop\TravisLedo-LidarScanRender-44895d9\TravisLedo-LidarScanRender-44895d9\LidarScanRender\nbproject\build-impl.xml:1330: The following error occurred while executing this line: C:\Users\Admin\Desktop\TravisLedo-LidarScanRender-44895d9\TravisLedo-LidarScanRender-44895d9\LidarScanRender\nbproject\build-impl.xml:936: Java returned: 1 BUILD FAILED (total time: 0 seconds)`
> 
> Only this error now, all the other errors look like it has been solved by your solution!

Aw man this the error that the other person asked about. I was never able to replicate it so not sure how to fix it.
Can you try these solutions and see if it works? Or other google searches for this error. It could be as simple as changing the JavaFX version.
https://stackoverflow.com/questions/53795661/javafx-modular-application-java-lang-module-findexception-module-javafx-contro


Edit: Saw your update lol. Did the errors come back after you build again?
Random thing to try as well. Maybe create a new project and slowly add each of the files/libraries from this project. They are all just simple java classes, so in theory should work? xD That way we could eliminate that it's a project problem but more of a library or jdk problem.

**Comment by yocarry** [comment]
I'm trying to debug this myself at the moment, and it seems I fixed a lot of problems (Mostly the PATH variables to the JavaFX libraries) just one error now:
``Error occurred during initialization of boot layer
java.lang.module.FindException: Module javafx.controls not found
C:\Users\Admin\Desktop\TravisLedo-LidarScanRender-44895d9\TravisLedo-LidarScanRender-44895d9\LidarScanRender\nbproject\build-impl.xml:1330: The following error occurred while executing this line:
C:\Users\Admin\Desktop\TravisLedo-LidarScanRender-44895d9\TravisLedo-LidarScanRender-44895d9\LidarScanRender\nbproject\build-impl.xml:936: Java returned: 1
BUILD FAILED (total time: 1 second)``

**Comment by yocarry** [comment]
`Error: Could not find or load main class lidarscanrender.Main
Caused by: java.lang.NoClassDefFoundError: javafx/application/Application
C:\Users\Admin\Desktop\TravisLedo-LidarScanRender-44895d9\TravisLedo-LidarScanRender-44895d9\LidarScanRender\nbproject\build-impl.xml:1330: The following error occurred while executing this line:
C:\Users\Admin\Desktop\TravisLedo-LidarScanRender-44895d9\TravisLedo-LidarScanRender-44895d9\LidarScanRender\nbproject\build-impl.xml:936: Java returned: 1
BUILD FAILED (total time: 0 seconds)`

This is the only error left; do you know any way to fix this?

**Comment by TravisLedo** [comment]
> Error: Could not find or load main class lidarscanrender.Main Caused by: java.lang.NoClassDefFoundError: javafx/application/Application

What OS are you using?
Have you tried this in Netbeans?

Right click on your Project in the project explorer
Click on properties
Click on Run
Make sure your Main Class is the one you want to be the entry point. (Make sure to use the fully qualified name i.e. mypackage.MyClass)
Click OK.
Clean an build your project
Run Project :)


If that does not work try closing Netbeans. Go to C:\Users\YOUR_PC_NAME\AppData\Local\Netbeans and delete the Cache folder. Then open the Netbeans again and run the project.

**Comment by yocarry** [comment]
I am using Windows 10
I am using this in Netbeans

The problem seems to be line 936 (`<java classname="@{classname}" dir="${work.dir}" failonerror="${java.failonerror}" fork="true">`) and line 1330 (`  <j2seproject1:java>`) in the build-impl.xml file

**Comment by TravisLedo** [comment]
Yea I think that it thinks the working directory main class file is still under my computer's path maybe? Did it ever ask you to select a main class when you tried to run it?

**Comment by yocarry** [comment]
It did ask me to select a main class

**Comment by TravisLedo** [comment]
> It did ask me to select a main class

I think the last thing I can think of is starting a new project and adding each java file one at a time and import the 2 libraries. That should work? lol

I also saw online that it may be easier if you install a jdk with JavaFX prepackaged. Try to create a hello world JavaFX project and see if you can even run that.

**Comment by TravisLedo** [comment]
@yocarry Hey, people were having issues with running the app like you and I just updated it to run with Gradle to make everyone's lives easier with the JavaFX library. Everything will auto import for you now and it should just work right away after Gradle downloads all the dependencies. I tested in IntelliJ so I recommend using that.

Let me know if that works for you and if you are not looking at this project anymore I will just close the issue. Thanks!
